codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -182,20 +236,55 @@ let updatePartition = (self: t, ~fetchState: FetchState.t, ~partitionId: partiti
   }
 }
 
+type nextQueries = WaitForNewBlock | NextQuery(array<FetchState.nextQuery>)
 /**
 Gets the next query from the fetchState with the lowest latestFetchedBlock number.
 */
-let getNextQuery = (self: t, ~eventFilters=?, ~currentBlockHeight) => {
-  let {fetchState, partitionId} = self->getMostBehindPartition
-
-  fetchState
-  ->FetchState.getNextQuery(~eventFilters?, ~currentBlockHeight, ~partitionId)
-  ->Result.map(((nextQuery, optUpdatesFetchState)) => (
-    nextQuery,
-    optUpdatesFetchState->Option.map(fetchState =>
-      self->updatePartition(~fetchState, ~partitionId)->Utils.unwrapResultExn
-    ),
-  ))
+let getNextQueriesOrThrow = (
+  self: t,
+  ~eventFilters=?,
+  ~currentBlockHeight,
+  ~maxPerChainQueueSize,
+  ~partitionsCurrentlyFetching,
+) => {
```

Comment:
This now returns an array of queries to make based on max concurrency, partitions currently fetching and which partition is most behind